src/app/components/client/TelemetryButton.tsx

```diff
@@ -0,0 +1,39 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+"use client";
+
+import { Button, ButtonProps } from "./Button";
+import { useTelemetry } from "../../hooks/useTelemetry";
+import { GleanMetricMap } from "../../../telemetry/generated/_map";
+
+export const TelemetryButton = <
+  EventModule extends keyof GleanMetricMap,
+  EventName extends keyof GleanMetricMap[EventModule],
+>(
+  props: {
+    event: {
+      module: EventModule;
+      name: keyof GleanMetricMap[EventModule];
+      data: GleanMetricMap[EventModule][EventName];
+    };
+  } & ButtonProps,
+) => {
+  const { event, onPress } = props;
+  const { module, name, data } = event;
+  const recordTelemetry = useTelemetry();
+
+  return (
+    <Button
+      {...props}
+      onPress={(e) => {
+        /* c8 ignore next 3 */
```

Comment:
I'm assuming this is the reason it's ignored for now, but if not, adding the real reason as a comment would be good :)

```suggestion
        // Consumers of this component do not currently have tests that actually click the button
        /* c8 ignore next 3 */
```

Reply:
Yup, that’s the reason. Added in [4e3f91e](https://github.com/mozilla/blurts-server/pull/4042/commits/4e3f91e504152b7a5e0d2f1c0af557c14b69fec8).